### PR TITLE
chore: Adds the filter type attribute to DataFilter model

### PR DESCRIPTION
**Pull request by CrisTofani**

### CODE REVIEW

File: src/types/filtering/filter.ts [hunk 1]
```diff
@@ -37,12 +44,17 @@ const DataFilterNullOrUndefined = t.intersection([
       t.literal("isNotUndefined"),
     ]),
     staticValue: t.union([t.undefined, t.never]),
-    compareField: t.union([t.undefined, t.never]),
   }),
 ]);
 
+const DataFilterFieldCommon = t.type({
+  type: t.literal("dynamicFilter"),
```
Comment (AleDore) [inline src/types/filtering/filter.ts:51]:
```suggestion
  filterType: t.literal("DYNAMIC"),
```

File: src/types/filtering/filter.ts [hunk 2]
```diff
@@ -6,6 +6,11 @@ const DataFilterCommon = t.type({
   fieldName: NonEmptyString,
 });
 
+const DataFilterStaticCommon = t.type({
+  type: t.literal("staticFilter"),
```
Comment (AleDore) [inline src/types/filtering/filter.ts:10]:
```suggestion
  filterType: t.literal("STATIC"),
```

File: src/types/filtering/filter.ts (hunk 1 above)
Comment (CrisTofani) [reply src/types/filtering/filter.ts]: changed in [94dfe00](https://github.com/pagopa/data-indexer-commons/pull/13/commits/94dfe00f13b3627cfac84c4908a701845eea702b)

File: src/types/filtering/filter.ts (hunk 2 above)
Comment (CrisTofani) [reply src/types/filtering/filter.ts]: as above

**Review by AleDore — APPROVED** [review]
lgtm

**Pull request merged**